firmware/raspberry_pi4_network_boot_beta.md

```diff
@@ -0,0 +1,119 @@
+# Raspberry Pi4 network boot (BETA)
+The page describes how to install and configure network boot for the Raspberry Pi4 Model B.
+
+**This is an early beta test and has only been tested on a small number of networks. Check with your network administrator before enabling network boot.**
+
+This documentation will be merged into the main bootloader documents after the beta test is complete.
+
+## Server configuration
+Network boot requires a TFTP and NFS server to be configured.  See [Network boot server tutorial](https://www.raspberrypi.org/documentation/hardware/raspberrypi/bootmodes/net_tutorial.md)
+
+Additional notes:-
+* The VC firmware (start4.elf) must be updated to the latest rpi-update version in order to get the Pi4 network drivers.
+* The MAC address on the Pi4 is programmed at manufacture and is not derived from the serial number.
+
+```
+# mac adddress (ip addr) - it should start with dc:a6:3
+ip addr | grep ether | head -n1 | awk '{print $2}' | tr [a-z] [A-Z]
+
+# serial number
+vcgencmd otp_dump | grep 28: | sed s/.*://g
+```
+
+## Installation - Updating firmware and bootloader
+```
+# Install the rpi-eeprom update package
+sudo apt update
+sudo apt upgrade
+sudo apt install rpi-eeprom
+
+# Get the latest VC firmware which includes the Pi4 network driver in start.elf
+sudo rpi-update
+
+# (Optional) Update rpi-eeprom-update to automatically get updates / bug fixes.
+sudo echo FIRMWARE_RELEASE_STATUS="beta" > /etc/default/rpi-eeprom-update
+```
+
+### Configuration - Enable network boot
+Network boot is not enabled by default in the bootloader. To enable it the bootloader configuration file must be edited.
+```
+# Extract the configuration file
+cp /lib/firmware/raspberrypi/beta/pieeprom-2019-09-22.bin pieeprom.bin
+rpi-eeprom-config pieeprom.bin > bootconf.txt
+
+# Enable network boot
+# change boot_order from 0x1 (sd-boot) to 
+BOOT_ORDER=0x21
+# sd-boot then failover to network boot
+
+# If you have a UART cable then setting BOOT_UART=1 will help debug network boot issues
+BOOT_UART=1
+
+# Apply the configuration change to the EEPROM image file
+rpi-eeprom-config --out pieeprom-netboot.bin --config bootconf.txt pieeprom.bin
+```
+
+### Update the bootloader EEPROM
+```
+# Flash the bootloader EEPROM with network boot enabled
+# Run 'rpi-eeprom-update -h' for more information
+sudo rpi-eeprom-update -d -f ./pieeprom-netboot.bin
+sudo reboot
+```
+
+## TFTP prefix
+As with the Pi3 bootloader the presence of /serial_number/start.elf is used to detect
+whether the serial number should be used as the directory prefix for TFTP requests.
+
+The bootloader loads start4.elf in preference to start.elf so it's possible to
+just have an empty start.elf file if you only want to have Pi4 firmware in the TFTP
+root directory.
+
```

Comment:
start4.elf is only used on the Pi 4 - is that right? And the Pi 4 doesn't use start.elf at all, except for this check for a serial number-specific directory on the TFTP server - is that right?